completion.go

```diff
@@ -106,3 +110,68 @@ func (c *Client) CreateCompletion(
 	err = c.sendRequest(req, &response)
 	return
 }
+
+// CreateCompletionStream — API call to create a completion w/ streaming
+// support. It sets whether to stream back partial progress. If set, tokens will be
+// sent as data-only server-sent events as they become available, with the
+// stream terminated by a data: [DONE] message.
+func (c *Client) CreateCompletionStream(
+	ctx context.Context,
+	request CompletionRequest,
+) (ch chan string, err error) {
+	var headerData = []byte("data: ")
+	var done = []byte("[DONE]")
+	var reqBytes []byte
+
+	request.Stream = true
+	reqBytes, err = json.Marshal(request)
+	if err != nil {
+		return
+	}
+
+	urlSuffix := "/completions"
+	req, err := http.NewRequest("POST", c.fullURL(urlSuffix), bytes.NewBuffer(reqBytes))
+	if err != nil {
+		return
+	}
+
+	req = req.WithContext(ctx)
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return
+	}
+	defer res.Body.Close()
+
+	reader := bufio.NewReader(req.Body)
+	var line []byte
+
+	ch = make(chan string)
+
+	go func() {
```

Comment:
There's a [recommendation](https://github.com/golang/go/wiki/CodeReviewComments#synchronous-functions) to prefer synchronous functions (also see Concurrency section [here](https://medium.com/@cep21/aspects-of-a-good-go-library-7082beabb403)).

Could we remove the goroutine from here? We can try something similar to the goroutine-free API proposal [here](https://github.com/sashabaranov/go-gpt3/issues/49#issuecomment-1373423407).

Reply:
It might also make sense to move all streaming-related logic into separate file

Reply:
Yes, that makes complete sense (just learned something new  lol). I will refactor per your guidelines. Thank you for your assistance.